src/conditions/datadog_search.rs

```diff
@@ -0,0 +1,749 @@
+            (
+                "NOT a:[50 TO *]",
+                log_event!["tags" => vec!["a:40"]],
+                log_event!["tags" => vec!["a:6"]],
+            ),
+            // Range - numeric, inclusive, unbounded (upper), tag (negate w/-).
+            (
+                "-a:[50 TO *]",
+                log_event!["tags" => vec!["a:40"]],
+                log_event!["tags" => vec!["a:6"]],
+            ),
+            // Range - numeric, inclusive, unbounded (lower), tag.
+            (
+                "a:[* TO 50]",
+                log_event!["tags" => vec!["a:400"]],
+                log_event!["tags" => vec!["a:600"]],
+            ),
+            // Range - numeric, inclusive, unbounded (lower), tag (negate).
+            (
+                "NOT a:[* TO 50]",
+                log_event!["tags" => vec!["a:600"]],
+                log_event!["tags" => vec!["a:400"]],
+            ),
+            // Range - numeric, inclusive, unbounded (lower), tag (negate w/-).
+            (
+                "-a:[* TO 50]",
+                log_event!["tags" => vec!["a:600"]],
+                log_event!["tags" => vec!["a:400"]],
+            ),
+            // Range - numeric, inclusive, unbounded (both), tag.
+            (
+                "a:[* TO *]",
+                log_event!["tags" => vec!["a:test"]],
+                log_event!["tags" => vec!["b:test"]],
+            ),
+            // Range - numeric, inclusive, unbounded (both), tag (negate).
+            (
+                "NOT a:[* TO *]",
+                log_event!["tags" => vec!["b:test"]],
+                log_event!["tags" => vec!["a:test"]],
+            ),
+            // Range - numeric, inclusive, unbounded (both), tag (negate w/-).
+            (
+                "-a:[* TO *]",
+                log_event!["tags" => vec!["b:test"]],
+                log_event!["tags" => vec!["a:test"]],
+            ),
+            // Range - numeric, inclusive, facet.
+            (
+                "@b:[1 TO 10]",
+                log_event!["custom" => json!({"b": 5})],
+                log_event!["custom" => json!({"b": 11})],
+            ),
+            // Range - numeric, inclusive, facet (negate).
+            (
+                "NOT @b:[1 TO 10]",
+                log_event!["custom" => json!({"b": 11})],
+                log_event!["custom" => json!({"b": 5})],
+            ),
+            // Range - numeric, inclusive, facet (negate w/-).
+            (
+                "-@b:[1 TO 10]",
+                log_event!["custom" => json!({"b": 11})],
+                log_event!["custom" => json!({"b": 5})],
+            ),
+            // Range - alpha, inclusive, facet.
+            (
+                "@b:[a TO z]",
+                log_event!["custom" => json!({"b": "c"})],
+                log_event!["custom" => json!({"b": 5})],
+            ),
+            // Range - alphanumeric, inclusive, facet.
+            (
+                r#"@b:["1" TO "100"]"#,
```

Comment:
Here we should also support `@b:[1 TO 100]` - the same logic as `@b:["1" TO "100"]` provided `b` is a string value. (Sorry, I should've emphasised this difference better).

Reply:
That should actually be the case already 👍🏻

Reply:
To be clear:

```
@b:[1 TO 10]
```

Is translated to:

```
(((((is_integer(.custom.b) || is_float(.custom.b)) && .custom.b >= 1) || .custom.b >= "1") && (((is_integer(.custom.b) || is_float(.custom.b)) && .custom.b <= 10) || .custom.b <= "10")) ?? false)
```

Is that right?

Reply:
can we add a test case with `@b:[1 TO 100]` to make sure it works as expected?